### ISSUE TITLE
cmd/go: 'go get' of non-main packages outside of a module is confusing in module mode

### ISSUE THREAD

**Issue by bcmills** [golang/go#47290]
In module mode today, `go get` on a non-`main` package downloads that package to the module cache, builds that package (until we finish #43684), and leaves it in the build cache.

This behavior can be confusing for users who are working from older, GOPATH-based materials (as in https://stackoverflow.com/q/68420141). In GOPATH mode this would have cloned the package's source repo to `GOPATH/src`, but in module mode it instead downloads the module containing the source to a versioned path within `GOPATH/pkg/mod`.

If `go get` is invoked on a non-`main` package in module mode outside of any module, we should probably at least log a message indicating where the source code is now found.

```
/tmp$ go1.17beta1 env GOMOD
/dev/null

/tmp$ go1.17beta1 get golang.org/x/net/html

/tmp$ find $(go1.17beta1 env GOPATH)/src -type f

/tmp$ go clean -modcache

/tmp$ go1.17beta1 get golang.org/x/net/html
go: downloading golang.org/x/net v0.0.0-20210716203947-853a461950ff

/tmp$
```

CC @jayconrod @matloob

**Comment by ianlancetaylor** [comment]
@bcmills This is in the 1.18 milestone; time to move to 1.19?  Thanks.

**Comment by ianlancetaylor** [comment]
@bcmills @matloob This issue is marked for 1.19.  It's just been rolling forward in milestones.  Should it move to Backlog?

**Comment by ianwoolf** [comment]
Now, in 1.19, if `go get` is invoked in module mode outside of any module, it will do nothing.

https://cs.opensource.google/go/go/+/refs/tags/go1.19:src/cmd/go/internal/modget/get.go;l=295

Seems like no other messages are needed anymore? Or am I missing something.  If something needs to be done, I'm happy to do it

**Comment by bcmills** [comment]
@ianwoolf, thanks for the ping. It does appear that this is fixed as of Go 1.19:
```
$ go1.19.3 env GOMOD
/dev/null

$ go1.19.3 get golang.org/x/net/html
go: go.mod file not found in current directory or any parent directory.
        'go get' is no longer supported outside a module.
        To build and install a command, use 'go install' with a version,
        like 'go install example.com/cmd@latest'
        For more information, see https://golang.org/doc/go-get-install-deprecation
        or run 'go help get' or 'go help install'.
```
